core/arch/arm/include/mm/core_mmu.h

```diff
@@ -218,6 +218,10 @@ struct core_mmu_phys_mem {
 		__register_memory_ul(#addr, (type), (addr), (size), \
 				     phys_mem_map)
 
+#define register_phys_mem_pgdir(type, addr, size) \
+		register_phys_mem(type, ROUNDDOWN(addr, CORE_MMU_PGDIR_SIZE), \
+				  ROUNDUP(size, CORE_MMU_PGDIR_SIZE))
```

Comment:
What if `addr + size` crosses a pgdir boundary? I guess you need to use:
`size + addr - ROUNDDOWN(addr, CORE_MMU_PGDIR_SIZE)`
instead of just `size`

Reply:
Yes, sounds correct. Will fix.